lib/application/widgets/material/decorated_tab_bar.dart

```diff
@@ -0,0 +1,30 @@
+import 'package:flutter/material.dart';
+
+/// Decorates a [TabBar] with custom layout specs
+class DecoratedTabBar extends StatelessWidget {
+  const DecoratedTabBar({
+    required this.tabBar,
+    required this.indicatorHeight,
+    required this.indicatorColor,
+    Key? key,
+  }) : super(key: key);
+
+  final TabBar tabBar;
+
+  /// Height for the bottom indicator line that crosses element's width
+  final double indicatorHeight;
+
+  /// Color for the bottom indicator line that crosses element's width
+  final Color indicatorColor;
+
+  @override
+  Widget build(BuildContext context) {
+    final bottomInsetBorder = Container(height: indicatorHeight, color: indicatorColor);
+    return Stack(
+      children: [
+        Positioned(bottom: 0, left: 0, right: 0, child: bottomInsetBorder),
```

Comment:
What do you think about the usage of an extension of `Positioned` which has `top`, `left`, `right` and `bottom` "sticky" constructors? Example:

```dart
extension PositionedWidget on Widget {
  /// Returns the Widget wrapped by a [Positioned] filling the vertical axis and positioned to the left of the [Stack]
  Positioned leftPositioned() => Positioned(left: 0, top: 0, bottom: 0, child: this);

  /// Returns the Widget wrapped by a [Positioned] filling the horizontal axis and positioned to the top of the [Stack]
  Positioned topPositioned() => Positioned(left: 0, top: 0, right: 0, child: this);

  /// Returns the Widget wrapped by a [Positioned] filling the vertical axis and positioned to the right of the [Stack]
  Positioned rightPositioned() => Positioned(top: 0, right: 0, bottom: 0, child: this);

  /// Returns the Widget wrapped by a [Positioned] filling the horizontal axis and positioned to the bottom of the
  /// [Stack]
  Positioned bottomPositioned() => Positioned(left: 0, right: 0, bottom: 0, child: this);
}
```

Reply:
I think they are valid in scenarios where we have to constantly deal with `Stack`, and it looks like it's not now the moment to do this extension. You could think for adding an utility to the main `layoutr` library, which would benefit from it, although this concept of common UI utilities is not introduced there yet.